flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/connectors/hive/read/HivePartitionFetcherContextBase.java

```diff
@@ -139,16 +138,9 @@ public List<ComparablePartitionValue> getComparablePartitionValueList() throws E
                                 tablePath.getDatabaseName(),
                                 tablePath.getObjectName(),
                                 Short.MAX_VALUE);
-                List<String> newNames =
-                        partitionNames.stream()
-                                .filter(
-                                        n ->
-                                                !partValuesToCreateTime.containsKey(
-                                                        extractPartitionValues(n)))
-                                .collect(Collectors.toList());
                 List<Partition> newPartitions =
                         metaStoreClient.getPartitionsByNames(
-                                tablePath.getDatabaseName(), tablePath.getObjectName(), newNames);
+                                tablePath.getDatabaseName(), tablePath.getObjectName(), partitionNames);
```

Comment:
Then it will always contain the all paritions, finally return all partitions with create time, right?
I prefer to only return the partitions that dose changes. Maybe you can compare the modifcation time for all partitions to decide which partition is updated and should be returned.

Reply:
OK, I'll look at your reply tonight. thanks

Reply:
@luoyuxia Hi, this is a great idea. However, this method getComparablePartitionValueList() returns all partitions in the current directory. The comparison and selection of the required partition information is done in the outer layer of the method. In addition, this method does not recommend directly returning the required partition information. Because in the case of bounded flow, all partition information is obtained by this method.